pydantic_ai_slim/pydantic_ai/_utils.py

```diff
@@ -31,11 +32,10 @@
 
 
 async def run_in_executor(func: Callable[_P, _R], *args: _P.args, **kwargs: _P.kwargs) -> _R:
-    if kwargs:
-        # noinspection PyTypeChecker
-        return await asyncio.get_running_loop().run_in_executor(None, partial(func, *args, **kwargs))
-    else:
-        return await asyncio.get_running_loop().run_in_executor(None, func, *args)  # type: ignore
+    loop = asyncio.get_running_loop()
+    ctx = contextvars.copy_context()  # copy the current context to the new thread
+    func_call = partial(func, *args, **kwargs)
+    return await loop.run_in_executor(None, ctx.run, func_call)
```

Comment:
I think we need to run this logic with `anyio`, they already do this underneath.